### ISSUE TITLE
Inline editing needs better keyboard handling

### ISSUE THREAD

**Issue by spincycle** [reactabular/reactabular#288]
react-edit supports TAB and ENTER out of the box. ENTER works as you would expect: You click on a cell, change its value, then hit ENTER and it saves. But that is also the behavior of TAB, which I would argue is not what the user expects. I expect the value to save and the cursor to move to the next column to the right, with an editor already active. 

The next logical step would be to support arrow keys, moving up and down and left and right, similar to how a spreadsheet works. There should be a visual cue, like a 1-pixel border around the cell, to show the user what cell currently has focus. From any selected cell, hitting ENTER would activate an editor. 

So in summary, my suggestions are:

1. Add concept of cell focus, by default would be a cell border that moves around the grid as the user hits UP, DOWN, LEFT, and RIGHT. Arrow keys never activate an editor. They just move focus.

1. ENTER activates an editor on any editable cell that has focus.

1. TAB behaves like RIGHT when not editing, behaves like ENTER + RIGHT + ENTER when editing.

**Comment by bebraw** [comment]
Yeah, the current TAB behavior is a little weird. I also agree support for arrow keys would be nice although that feels like a separate helper that would hook into the concept of cell focus somehow. 

It's quite possible all of this functionality will grow within [react-edit](https://github.com/reactabular/react-edit) as there isn't anything that's tried to reactabular in the requirements.

You can get inspiration for a helper from [selectabular](https://github.com/reactabular/selectabular#selecting-by-arrow-keys). You'll end up with something like `edit.byArrowKeys` that encapsulates the selection concern. You also need some way to connect that with the editable cells (harder problem) although that might go through the concept of editing state. Maybe you'll expose a callback where the user can decide and that's it.

**Comment by crimsonsunset** [comment]
Is there any movement on this? I've been hacking together tab key functionality I'm entirely unhappy with. Would love to have [optional?] support for the tab key natively through react-edit as described above .

**Comment by bebraw** [comment]
Nah, I don't do any active development on Reactabular anymore due to lack of funding.

That said, I can accept and review a PR if someone is willing to contribute.
